### PR TITLE
ENH: Add convenience function ReadImage

### PR DESCRIPTION
**Pull request by N-Dekker**
Added `itk::ReadImage<TOutputImage>(const std::string & filename)`, to
allows simplifying the C++ code for reading an image.

Suggested by Dženan Zukić (@dzenanz) at
https://github.com/InsightSoftwareConsortium/ITK/pull/2102#issuecomment-723232937

Follow-up to:
"ENH: add a convenience function WriteImage"
Pull request: #2160
Commit: 60b0984d4196841844424e33963e161f3f0c709d

Replaced manual creation of `ImageFileReader` objects in Common tests by
calls to the new `itk::ReadImage` function. Also removed related
redundant try-catch blocks from SymmetricSecondRankTensorImage tests.

### CODE REVIEW

**Comment by N-Dekker** [comment]
Review suggestion: ignore white space when comparing the files: https://github.com/InsightSoftwareConsortium/ITK/pull/2184/files?w=1

**Comment by hjmjohnson** [comment]
Nice!  LGTM, but I think @dzenanz should have a look before merging.

**Comment by N-Dekker** [comment]
@hjmjohnson  Thank you for your approval, Hans. An alternative could be to add the function as a `static`member function to `ImageFileReader`. In order to support a call like:

     auto image = itk::ImageFileReader<TOutputImage>::ReadImage(filename);

That would also be OK to me. But I think people prefer to just call `itk::ReadImage<TOutputImage>(filename)`  😃

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
@thewtex we should definitely mention these convenience functions in 5.2.0 release notes.

File: Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx
Comment (blowekamp) [inline Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx:47]: This is a change in the behavior of this test. This test is called "streaming" because it test streaming. This change update the whole image from a file then runs the filter.

File: Modules/IO/ImageBase/include/itkImageFileReader.h
Comment (blowekamp) [inline Modules/IO/ImageBase/include/itkImageFileReader.h:172]: This description seems short.

A description should include that the function updated the whole image. Also perhaps a description of the template parameters or the behaviors of the pixel trains template parameter would be good.

**Review by blowekamp — CHANGES_REQUESTED** [review]
The behavior to the streaming test changes with this PR.

File: Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx
Comment (N-Dekker) [reply Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx:47]: OK, thanks Bradley, but would that be a problem? The test still passes successfully, right?

File: Modules/IO/ImageBase/include/itkImageFileReader.h
Comment (N-Dekker) [reply Modules/IO/ImageBase/include/itkImageFileReader.h:172]: > A description should include that the function updated the whole image.

Thanks again @blowekamp but when a user does `auto image = itk::ReadImage(fileName)`, it seems obvious to me that the whole image is updated. A user cannot do something like `image->Update()` afterwards anyway. OK?

I do agree with you that a description of the template parameter `ConvertPixelTraits` would be good. I'll add one with the next force-push. 😃

File: Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx
Comment (dzenanz) [reply Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx:47]: The test passes successfully, but its whole point is to test streaming. I did not catch this in my review.

File: Modules/IO/ImageBase/include/itkImageFileReader.h
Comment (blowekamp) [reply Modules/IO/ImageBase/include/itkImageFileReader.h:172]: I may be wrong, but I thought one of the ITK python procedural methods return objects ( Image or Filter?) that still needed to be updated?

I think executing `image->Update()` afterwords would still work. While its not this case, in others, this would have the side effect of setting the LargestPossibleRegion to the BufferedRegion.

File: Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx
Comment (N-Dekker) [reply Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx:47]: OK, I will remove the proposed itkStreamingImageFilterTest3 changes from this PR, with the next force-push. Thanks @blowekamp and @dzenanz

File: Modules/IO/ImageBase/include/itkImageFileReader.h
Comment (N-Dekker) [reply Modules/IO/ImageBase/include/itkImageFileReader.h]: OK, I did not even know that `image->Update()` would work. But then again, isn't it obvious to you that the image returned by `ReadImage` is the image that was just read?

Comment (dzenanz) [reply Modules/IO/ImageBase/include/itkImageFileReader.h]: It is obvious. The docstring is for people who are having some doubts, maybe because they know streaming exists in ITK 😄 and want to check whether this convenience function participates.

File: Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx
Comment (hjmjohnson) [reply Modules/Core/Common/test/itkStreamingImageFilterTest3.cxx]: Thanks @N-Dekker

File: Modules/IO/ImageBase/include/itkImageFileReader.h
Comment (N-Dekker) [reply Modules/IO/ImageBase/include/itkImageFileReader.h]: @blowekamp @dzenanz Just extended the function description, hope you appreciate 😃 I added the following sentence:

> The function reads the image from the specified file, and returns the image that it has read.

https://github.com/InsightSoftwareConsortium/ITK/blob/3778f98e1f4d2edc53c135d8526ed2ab3da665d5/Modules/IO/ImageBase/include/itkImageFileReader.h#L178-L179

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Do we need `ITK_TEMPLATE_EXPORT` in the declaration? Sorry I noticed this just now.

**Pull request merged**
